app/services/CoreRegistrationValidationService.scala

```diff
@@ -40,4 +40,41 @@ class CoreRegistrationValidationService @Inject()(connector: ValidateCoreRegistr
       case _ => None
     }
   }
+
+  def searchEuTaxId(euTaxReference: String, countryCode: String)
+                   (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Match]] = {
+
+    val coreRegistrationRequest = CoreRegistrationRequest(SourceType.EUTraderId.toString, None, euTaxReference, None, countryCode)
+
+    connector.validateCoreRegistration(coreRegistrationRequest).map {
+
+      case Right(coreRegistrationResponse) => coreRegistrationResponse.matches.headOption
+
+      case _ => None
+    }
+  }
+
+  def searchEuVrn(euVrn: String, countryCode: String)
+                 (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Match]] = {
+
+    val coreRegistrationRequest = CoreRegistrationRequest(SourceType.EUVATNumber.toString, None, euVrn, None, countryCode)
+
+    connector.validateCoreRegistration(coreRegistrationRequest).map {
+
+      case Right(coreRegistrationResponse) => coreRegistrationResponse.matches.headOption
+
+      case _ => None
+    }
+  }
+
+  def isActiveTrader(activeMatch: Match): Boolean = {
```

Comment:
I like this refactor, can we add some unit tests for these two functions?

Reply:
Unit tests added for new added functions.